src/porepy/numerics/interface_laws/elliptic_interface_laws.py

```diff
@@ -723,4 +725,260 @@ def assemble_matrix_rhs(
+                matrix[slave_ind, slave_ind].shape[1],
+                self.ndof(mg),
+            ]
+        )
+        cc = np.array([sps.coo_matrix((i, j)) for i in dof for j in dof])
+        cc_master = cc.reshape((3, 3))
+        cc_slave = cc_master.copy()
+
+        # The rhs is just zeros
+        # EK: For some reason, the following lines were necessary to apease python
+        rhs = np.empty(3, dtype=np.object)
+        rhs[master_ind] = np.zeros(matrix[master_ind, master_ind].shape[1])
+        rhs[slave_ind] = np.zeros(matrix[slave_ind, slave_ind].shape[1])
+        rhs[2] = np.zeros(self.ndof(mg))
+
+        # The convention, for now, is to put the master grid information
+        # in the first column and row in matrix, slave grid in the second
+        # and mortar variables in the third
+        # If master and slave is the same grid, they should contribute to the same
+        # row and coloumn. When the assembler assigns matrix[idx] it will only add
+        # the slave information because of duplicate indices (master and slave is the same).
+        # We therefore write the both master and slave info to the slave index.
+        if g_master == g_slave:
+            master_ind = 1
+        else:
+            master_ind = 0
+
+        # Obtain the contribution of the cell centered pressure on the displacement
+        # trace u_master
+        self.discr_master.assemble_int_bound_displacement_trace(
+            g_master, data_master, data_edge, False, cc_master, matrix, master_ind
+        )
+        # Equivalent for u_slave
+        self.discr_slave.assemble_int_bound_displacement_trace(
+            g_slave, data_slave, data_edge, True, cc_slave, matrix, slave_ind
+        )
+        # We now have to flip the sign of some of the matrices
+        # First we flip the sign of the master stress because the mortar stress
+        # is defined from the slave stress. Then, stress_master = -\lambda
+        cc_master[master_ind, 2] = -cc_master[master_ind, 2]
+        # Then we flip the sign for the master displacement since the displacement
+        # jump is defined as u_slave - u_master
+        cc_master[2, master_ind] = -cc_master[2, master_ind]
+
+        matrix += cc_master + cc_slave
+
+        # The displacement jump is scaled by a matrix in the Robin condition:
+        robin_weight = matrix_dictionary_edge["robin_weight"]
+
+        for i in range(3):
+            matrix[2, i] = robin_weight * matrix[2, i]
+
+        # The following two functions might or might not be needed when using
+        # a finite element discretization (see RobinCoupling for flow).
+        self.discr_master.enforce_neumann_int_bound(
+            g_master, data_edge, matrix, False, master_ind
+        )
+        self.discr_slave.enforce_neumann_int_bound(
+            g_slave, data_edge, matrix, True, slave_ind
+        )
+
+        return matrix, rhs
+
+
+class DivU_StressMortar(RobinContactBiotPressure):
```

Comment:
Is mpsa.py the right module for this function? Not that I have an immediate better suggestion.

Reply:
this is the elliptic_interface_law_module. Not sure if this is the right one either